mmcls/models/backbones/gcvit.py

```diff
@@ -0,0 +1,792 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+# by lihao
+# Reproduction for GCViT
+
+import torch
+import torch.nn as nn
+from mmengine.model.weight_init import trunc_normal_
+
+from mmcls.registry import MODELS
+
+
+def drop_path(x,
+              drop_prob: float = 0.,
+              training: bool = False,
+              scale_by_keep: bool = True):
+
+    if drop_prob == 0. or not training:
+        return x
+    keep_prob = 1 - drop_prob
+    shape = (x.shape[0], ) + (1, ) * (
+        x.ndim - 1)  # work with diff dim tensors, not just 2D ConvNets
+    random_tensor = x.new_empty(shape).bernoulli_(keep_prob)
+    if keep_prob > 0.0 and scale_by_keep:
+        random_tensor.div_(keep_prob)
+    return x * random_tensor
+
+
+class DropPath(nn.Module):
+    """Drop paths (Stochastic Depth) per sample  (when applied in main path of
+    residual blocks)."""
+
+    def __init__(self, drop_prob: float = 0., scale_by_keep: bool = True):
+        super(DropPath, self).__init__()
+        self.drop_prob = drop_prob
+        self.scale_by_keep = scale_by_keep
+
+    def forward(self, x):
+        return drop_path(x, self.drop_prob, self.training, self.scale_by_keep)
+
+    def extra_repr(self):
+        return f'drop_prob={round(self.drop_prob,3):0.3f}'
+
+
+def _to_channel_last(x):
+    """
+    Args:
+        x: (B, C, H, W)
+
+    Returns:
+        x: (B, H, W, C)
+    """
+    return x.permute(0, 2, 3, 1)
+
+
+def _to_channel_first(x):
+    """
+    Args:
+        x: (B, H, W, C)
+
+    Returns:
+        x: (B, C, H, W)
+    """
+    return x.permute(0, 3, 1, 2)
+
+
+def window_partition(x, window_size, h_w, w_w):
```

Comment:
@mzr1996 This function is equivalent to https://github.com/open-mmlab/mmclassification/blob/2037260ea6c98a3b115e97727e1151a1c2c32f7a/mmcls/models/utils/attention.py#L265 maybe we can use `ShiftWindowWSA.window_reverse` directly or move this function to the outside of the `ShiftWindowWSA`, would you like to share your comments?